src/apphosting/config.ts

```diff
@@ -55,12 +56,45 @@ export function yamlPath(cwd: string): string | null {
 }
 
 /** Load apphosting.yaml */
-export function load(yamlPath: string): Config {
+export function load(yamlPath: string): yaml.Document {
   const raw = fs.readFile(yamlPath);
-  return yaml.load(raw, yaml.DEFAULT_FULL_SCHEMA) as Config;
+  return yaml.parseDocument(raw);
 }
 
 /** Save apphosting.yaml */
-export function store(yamlPath: string, config: Config): void {
-  writeFileSync(yamlPath, yaml.dump(config));
+export function store(yamlPath: string, document: yaml.Document): void {
+  writeFileSync(yamlPath, document.toString());
+}
+
+export function getEnv(document: yaml.Document, variable: string): Env | undefined {
+  if (!document.has("env")) {
+    return undefined;
+  }
+  const envs = document.get("env") as yaml.YAMLSeq;
+  for (const env of envs.items as Array<NodeType<Env>>) {
+    if ((env.get("variable") as unknown) === variable) {
+      return env.toJSON() as Env;
+    }
+  }
+  return undefined;
+}
+
+export function setEnv(document: yaml.Document, env: Env): void {
+  if (!document.has("env")) {
+    document.set("env", document.createNode([env]));
+    return;
+  }
+  const envs = document.get("env") as yaml.YAMLSeq<NodeType<Env>>;
+
+  // The type system in this library is... not great at propagating type info
+  const envYaml = document.createNode(env);
+  for (let i = 0; i < envs.items.length; i++) {
+    if ((envs.items[i].get("variable") as unknown) === env.variable) {
```

Comment:
This is very nitpicky but line items in yaml are not guaranteed to be unique. This will replace the first instance of `env.variable` but will leave duplicates in place